src/pubsub/ua_pubsub.h

```diff
@@ -77,7 +77,8 @@ void
 UA_PubSubConnection_clear(UA_Server *server, UA_PubSubConnection *connection);
 /* Register channel for given connectionIdentifier */
 UA_StatusCode
-UA_PubSubConnection_regist(UA_Server *server, UA_NodeId *connectionIdentifier);
+UA_PubSubConnection_regist(UA_Server *server, UA_NodeId *connectionIdentifier, UA_ExtensionObject *transportSettings,
+                           void (*callback)(UA_ByteString *encodedBuffer, UA_ByteString *topic));
```

Comment:
The UA_PubSubConnection_regist is generic for all PubSub transport protocols. The callback should not be needed anymore after finishing the subscribe API and the term and parameter topic is very MQTT specific. We should try to avoid these changes.